packages/rs-drive/src/drive/identity/withdrawals/queue.rs

```diff
@@ -0,0 +1,186 @@
+    WithdrawalTransaction,
+};
+
+impl Drive {
+    /// Add insert operations for withdrawal transactions to the batch
+    pub fn add_enqueue_withdrawal_transaction_operations<'a>(
+        &self,
+        withdrawals: &'a [WithdrawalTransaction],
+        drive_operations: &mut Vec<DriveOperationType<'a>>,
+    ) {
+        if !withdrawals.is_empty() {
+            drive_operations.push(DriveOperationType::WithdrawalOperation(
+                WithdrawalOperationType::InsertTransactions {
+                    transactions: withdrawals,
+                },
+            ));
+        }
+    }
+
+    /// Get specified amount of withdrawal transactions from the DB
+    pub fn dequeue_withdrawal_transactions(
+        &self,
+        num_of_transactions: u16,
+        transaction: TransactionArg,
+    ) -> Result<Vec<WithdrawalTransaction>, Error> {
+        let mut query = Query::new();
+
+        query.insert_item(QueryItem::RangeFull(RangeFull));
+
+        let path_query = PathQuery {
+            path: get_withdrawal_transactions_queue_path(),
+            query: SizedQuery {
+                query,
+                limit: Some(num_of_transactions),
+                offset: None,
+            },
+        };
+
+        let result_items = self
+            .grove
+            .query_raw(
+                &path_query,
+                transaction.is_some(),
+                QueryResultType::QueryKeyElementPairResultType,
+                transaction,
+            )
+            .unwrap()
+            .map_err(Error::GroveDB)?
+            .0
+            .to_key_elements();
+
+        let withdrawals = result_items
+            .into_iter()
+            .map(|(id, element)| match element {
+                Element::Item(bytes, _) => Ok((id, bytes)),
+                _ => Err(Error::Drive(DriveError::CorruptedWithdrawalNotItem(
+                    "withdrawal is not an item",
+                ))),
+            })
+            .collect::<Result<Vec<(Vec<u8>, Vec<u8>)>, Error>>()?;
+
+        if !withdrawals.is_empty() {
+            let mut batch_operations: Vec<DriveOperation> = vec![];
+            let mut drive_operations: Vec<DriveOperation> = vec![];
+
+            let withdrawals_path: [&[u8]; 2] = get_withdrawal_transactions_queue_path_as_u8();
+
+            for (id, _) in withdrawals.iter() {
+                self.batch_delete(
+                    withdrawals_path,
+                    id,
+                    // we know that we are not deleting a subtree
+                    BatchDeleteApplyType::StatefulBatchDelete {
+                        is_known_to_be_subtree_with_sum: Some((false, false)),
+                    },
+                    transaction,
+                    &mut batch_operations,
+                )?;
+            }
+
+            self.apply_batch_drive_operations(
+                None,
+                transaction,
+                batch_operations,
+                &mut drive_operations,
+            )?;
```

Comment:
This is the wrong approach. You should see a batch as an atomic operation. In this case you could basically remove withdrawal transactions without updating the state of the associated withdrawals. If there was some error drive would be in a corrupted state. What you want to do instead is have these updates be batched with the updating of states of the associated withdrawals.